lib/Event/ReorderEvent.php

```diff
@@ -30,7 +32,7 @@ class ReorderEvent extends AbstractDocumentEvent
      */
     public function __construct($document, $destId, $after)
     {
-        parent::__construct($document);
+        $this->document = $document;
```

Comment:
There will only ever be a single implementation of any given event class. Introducing interfaces for the sake of the type hint that is already implemented is solving a problem that does not exist. If there is a valid case then lets consider this, but currently I do not see the problem. I would rather the code be not-perfect than it be prematurely over enginered,